src/boxed.rs

```diff
@@ -366,7 +366,7 @@ impl<T: 'static, MM: BoxedMemoryManager<T>> Uninitialized for Boxed<T, MM> {
     unsafe fn uninitialized() -> Self {
         Boxed {
             inner: {
-                let mut inner = Box::<T>::new(mem::zeroed());
+                let mut inner = Box::<T>::new(mem::MaybeUninit::zeroed().assume_init());
```

Comment:
This is wrong again and defeats the purpose of `MaybeUninit`. This code here has to be

```rust
let mut inner = Box::<T>::new(mem::MaybeUninit::zeroed();
MM::init(inner.as_mut_ptr());

AnyBox::Native(inner.assume_init());
```